platform/linux-generic/odp_ipsec.c

```diff
@@ -375,6 +375,34 @@ typedef struct {
 	uint8_t	iv[IPSEC_MAX_IV_LEN];
 } ipsec_state_t;
 
+/*
+ * Computes 64-bit seq number according to RFC4303 A2
+ */
+static inline uint64_t ipsec_compute_esn(ipsec_sa_t *ipsec_sa, uint32_t seq)
+{
+	uint32_t wintop_h, wintop_l, winbot_l, ws;
+	uint64_t seq64 = 0;
+
+	wintop_h = odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq) >> 32;
+	wintop_l = odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq) & 0xffffffff;
```

Comment:
You should not split the load to loads like that. The two loads could observe different values of the variable (i.e. a write by another thread occuring in between the loads).

Reply:
Ack, will use the load once and use this loaded value to get "wintop_h" and "wintop_l".